pkgs/native_toolchain_c/lib/src/cbuilder/linker_options.dart

```diff
@@ -109,33 +123,37 @@ extension LinkerOptionsExt on LinkerOptions {
     switch (targetOS) {
       case OS.macOS || OS.iOS:
         return [
-          if (!_includeAllSymbols) ...sourceFiles,
+          if (!_keepAllSymbols) ...sourceFiles,
           ..._toLinkerSyntax(tool, [
-            if (_includeAllSymbols) ...sourceFiles.map((e) => '-force_load,$e'),
+            if (_keepAllSymbols) ...sourceFiles.map((e) => '-force_load,$e'),
             ..._linkerFlags,
-            ..._symbolsToKeep?.map((symbol) => '-u,_$symbol') ?? [],
+            ..._symbols.map((symbol) => '-u,_$symbol'),
             if (stripDebug) '-S',
             if (gcSections) '-dead_strip',
+            if (_linkerScriptMode is ManualLinkerScript)
+              '-exported_symbols_list,${_linkerScriptMode.script.toFilePath()}',
+            if (_linkerScriptMode is GenerateLinkerScript)
```

Comment:
The Android/Linux and Windows branch uses `else if` here. Keep it consistent?

Reply:
sure